#52 Add full constuctor to generated "impl" classes

Kaiser1989 opened the pull request:
Added a constructor with all fields and adjusted corresponding tests of Objects. Constructor will only be generated for implementation classes. Additional properties will be ignored.
Also added an empty constructor for jackson geneneration and to not break current behaviour.

jpbelang commented:
Sorry, I've been stuck on the parse for a bit.

is there any way to make this a plugin ?

Kaiser1989 commented:
Yeah, i added it as plugin:
name: core.constructor

I also added some tests for empty classes, inherited classes, and simply classes.

It only works for objects:
- If object has a field (other than additionalProperties), empty and full-args constructor are added
- discriminator value is not added to constructor
- additionalProperties are not added to constructor

Best regards
